UPGRADING.md

```diff
@@ -1,14 +1,78 @@
 Upgrading
 =========
 
+## 3.x to 4.x
+
+#### Setting Endpoints
+
+The way endpoints will be configured has been change slightly, in order to make sure all of your requests get send to the correct place.  You'll no longer be able to pass the `endpoint` parameter to your `Bugsnag\Client::make` calls, and should instead use the [`setEndpoints` configuration option](https://docs.bugsnag.com/platforms/php/other/configuration-options/#endpoints).  If you've previously used the `BUGSNAG_ENDPOINT` environment variable, it's now been renamed to `BUGSNAG_NOTIFY_ENDPOINT`, and will only work in conjunction with the `BUGSNAG_SESSION_ENDPOINT` environment variable, to make sure that both endpoints are being used correctly.
+
+Old way:
+
+```
+$bugsnag = Bugsnag\Client::make("YOUR_API_KEY", "http://bugsnag-notify.example.com);
+$bugsnag->setSessionEndpoint("http://bugsnag-session.example.com);
+```
+
+New way:
+
+```
+$bugnsag = Bugsnag\Client::make("YOUR_API_KEY");
+$bugsnag->setEndpoints(
+    "http://bugsnag-notify.example.com",
+    "http://bugsnag-session.example.com"
+);
+```
+
+#### Configuring the GuzzleHttp Client
+
+If you've previously customized the GuzzleHttp Client we use to deliver notifications, you'll no longer be able to pass it in to a `new Bugsnag\Client` call. Instead you'll need to add it to Bugsnag using the [`setGuzzleClient` configuration option.](https://docs.bugsnag.com/platforms/php/other/configuration-options/#guzzle-client)
+
+Old way:
+
+```
+$bugsnag = new Bugsnag\Client(new Bugsnag\Configuration("YOUR_API_KEY"), null, $myCustomGuzzle);
+```
+
+New way:
+
+```
+$bugsnag = new Bugsnag\Client::make("YOUR_API_KEY");
+$bugsnag->setGuzzleClient($myCustomGuzzle);
+```
+
+#### Deprecation
```

Comment:
Why is this change required for always on session tracking?

Reply:
This isn't, but should be removed _if_ this is a major release.  I've removed it as it'll be a part of a release PR instead.